src/fsharp/tast.fs

```diff
@@ -35,12 +35,12 @@ open Microsoft.FSharp.Core.CompilerServices
 /// Unique name generator for stamps attached to lambdas and object expressions
 type Unique = int64
 //++GLOBAL MUTABLE STATE
-let newUnique = let mutable i = 0L in fun () -> System.Threading.Interlocked.Increment(&i)
+let newUnique = let i = ref 0L in fun () -> System.Threading.Interlocked.Increment(i)
```

Comment:
Mutable is ok. Warning is only informational
I disabled this warning ( converted to error ) in the project with nowarn